tests/fdb/api/test_archive_callback.cc

```diff
@@ -0,0 +1,73 @@
+#include "eckit/testing/Test.h"
+#include "fdb5/api/FDB.h"
+
+namespace fdb5::test {
+
+//----------------------------------------------------------------------------------------------------------------------
+CASE("Archive callback") {
+    FDB fdb;
+
+	std::string data_str = "Raining cats and dogs";
+    const void* data = static_cast<const void *>(data_str.c_str());
+    size_t length = data_str.size();
+
+	Key key;
+	key.set("class","od");
+	key.set("expver","xxxx");
+	key.set("type","fc");
+	key.set("stream","oper");
+	key.set("date","20101010");
+	key.set("time","0000");
+	key.set("domain","g");
+	key.set("levtype","sfc");
+	key.set("param","130");
+
+    std::map<fdb5::Key, eckit::URI> map;
+    std::vector<Key> internalKeys;
+
+    fdb.registerCallback([&map] (const fdb5::Key& internalKey, const fdb5::FieldLocation& location) {
+        map[internalKey] = location.fullUri();
+    });
+
+	key.set("step","1");
+    internalKeys.push_back(fdb.archive(key, data, length));
+
+	key.set("step","2");
+    internalKeys.push_back(fdb.archive(key, data, length));
+
+	key.set("step","3");
```

Comment:
Please make sure we have a test that checks changing metadata at level 1/2 of the schema as well. That has different internal impacts.